src/ert/simulator/batch_simulator.py

```diff
@@ -228,26 +228,28 @@ def start(
         time, so when you have called the 'start' method you need to let that
         batch complete before you start a new batch.
         """
-        with open_storage(self.ert_config.ens_path, "w") as storage:
-            experiment = storage.create_experiment()
-            ensemble = storage.create_ensemble(
-                experiment.id, name=case_name, ensemble_size=self.ert.getEnsembleSize()
-            )
-            self.ert.addDataKW("<CASE_NAME>", _slug(ensemble.name))
-            for sim_id, (geo_id, controls) in enumerate(case_data):
-                assert isinstance(geo_id, int)
-                self._setup_sim(sim_id, controls, ensemble)
-
-            # The input should be validated before we instantiate the BatchContext
-            # object, at that stage a job_queue object with multiple threads is
-            # started, and things will typically be in a quite sorry state if an
-            # exception occurs.
-            itr = 0
-            mask = [True] * len(case_data)
-            sim_context = BatchContext(
-                self.result_keys, self.ert, ensemble, mask, itr, case_data
-            )
-
-            if self.callback:
-                self.callback(sim_context)
-            return sim_context
+        storage = open_storage(self.ert_config.ens_path, "w")
```

Comment:
Removing the context manager means that `storage` is never `close`d. Is that a good idea? Is it at all possible to keep the context manager and wait for BatchContext to finish?

Reply:
Might not be a good idea, the reason I removed the context manager was because otherwise it goes out of scope before the callbacks are called, they run in a different thread. Do you have a suggestion for how to improve it?